data_model/src/trigger/set.rs

```diff
@@ -221,15 +240,60 @@ impl Set {
     /// Finds all actions, that are triggered by `event` and stores them.
     /// This actions will be inspected in the next [`Set::handle_data_event()`] call
     pub fn handle_data_event(&self, event: &DataEvent) {
-        self.handle_event(&self.data_triggers, event, EventType::Data)
+        let data_domain_id = match event {
+            DataEvent::Trigger(
+                TriggerEvent::Created(id)
+                | TriggerEvent::Deleted(id)
+                | TriggerEvent::Extended(id)
+                | TriggerEvent::Shortened(id),
+            ) => id.domain_id.clone(),
+            _ => None,
+        };
+
+        for entry in self.data_triggers.iter() {
+            let id = entry.key();
+
+            // Check only if the trigger is domain scoped
+            if id.domain_id.is_some() && id.domain_id == data_domain_id {
+                continue;
+            }
```

Comment:
```suggestion
            // Check only if the trigger is domain scoped
            if id.domain_id.is_some() && id.domain_id != data_domain_id {
                continue;
            }
```

Shouln't it be like this?